framework/src/node/consensus/network_endpoint.ts

```diff
@@ -167,6 +175,67 @@ export class NetworkEndpoint {
 		});
 	}
 
+	public handleEventSingleCommit(data: unknown, peerId: string): void {
+		this._addRateLimit(
+			NETWORK_RPC_GET_SINGLE_COMMIT_FROM_ID,
+			peerId,
+			DEFAULT_SINGLE_COMMIT_FROM_IDS_RATE_LIMIT_FREQUENCY,
+		);
+		let decodedData: SingleCommit;
+
+		try {
+			decodedData = codec.decode<SingleCommit>(singleCommitSchema, data as never);
```

Comment:
Other endpoints have data schema defined, so it would be better to have the surrounding schema. ex)
```
{
  singleCommit: Buffer
}
```
like in https://github.com/LiskHQ/lisk-sdk/blob/feature/6554-improve-framework-architecture/framework/src/node/consensus/schema.ts#L96-L107